indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskClient.java

```diff
@@ -376,10 +385,47 @@ public ListenableFuture<DateTime> getStartTimeAsync(final String id)
     return doAsync(() -> getStartTime(id));
   }
 
-
   public ListenableFuture<Map<PartitionIdType, SequenceOffsetType>> pauseAsync(final String id)
   {
-    return doAsync(() -> pause(id));
+    ListenableFuture<Map<PartitionIdType, SequenceOffsetType>> future = doAsync(() -> pause(id));
+    pausingTaskFutureMap.put(id, new TaskPauseControlInfo(future, true));
```

Comment:
```suggestion
    pausingTaskFutureMap.put(id, future);
```

Reply:
We should also add a callback on this future to remove entries from the map when the future completes. Otherwise, the map `pausingTaskFutureMap` will keep growing in size.

Reply:
yes, done